### PR TITLE
kconfig: move more XTOS-only options out from top-level Kconfig

### PR DESCRIPTION
**Pull request by kv2019i**
Some of the top-level Kconfig options are not used at all when SOF is built with another RTOS like the SOF Zephyr module build. 

To make this more explicit at Kconfig level, move these options out from the top-level Kconfig file and exclude when needed.

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -127,39 +127,9 @@ config XT_INTERRUPT_LEVEL_5
...
-
-config OPTIMIZE_FOR_NONE
-	bool "Don't optimize"
-	help
-	  Apply no compiler optimizations.
-	  It means -O0 for GCC or equivalent for other compilers.
-
-endchoice
+if !ZEPHYR_SOF_MODULE
+	rsource "Kconfig.xtos-build"
```
Comment (marc-hb) [inline Kconfig:131]: I fetched and tested this and it works but I don't understand how, can you add a comment explaining why `git grep ZEPHYR_SOF_MODULES` => nothing?

BTW https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#lack-of-conditional-includes

Comment (marc-hb) [reply Kconfig:131]: I found a few, auto-generated ZEPHYR_SOF_ variables but not this one  at https://docs.zephyrproject.org/latest/guides/modules.html

File: Kconfig.xtos-build [hunk 2]
```diff
@@ -35,3 +35,9 @@ config OPTIMIZE_FOR_NONE
 	  It means -O0 for GCC or equivalent for other compilers.
 
 endchoice
+
+config BUILD_VM_ROM
```
Comment (lgirdwood) [inline Kconfig.xtos-build:39]: I think this is needed for Zephyr as it's a ROM stub that jumps to SOF bootloader code in qemu.  Unless Zephyr has this jump built in when building a qemu target ?

Comment (kv2019i) [reply Kconfig.xtos-build:39]: @lgirdwood I was wondering the same, but this is not used in any of the Zephyr builds and the implementation is in xtensa specific XTOS code, so I moved this out. But let me check. @andyross and @lyakh -- is this used in in Zephyr (to boot in Qemu) via some other way I don't see.

File: Kconfig (hunk 1 above)
Comment (kv2019i) [reply Kconfig:131]: @marc-hb Working on a reply. I just observed this is defined in my build .config and used it (and it works). But let me dig into where it's actually defined.

Comment (marc-hb) [reply Kconfig:131]: Thanks! https://www.google.com/search?q=programming+by+coincidence :-)

File: Kconfig.xtos-build (hunk 2 above)
Comment (lyakh) [reply Kconfig.xtos-build:39]: IIRC to run Zephyr-generated SOF in Qemu I was using VMROM, built separately by a native SOF build. But this was a few months ago so I might be wrong

**Review by lyakh — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

File: Kconfig (hunk 1 above)
Comment (kv2019i) [reply Kconfig:131]: @marc-hb  Zephyr's build system adds this for each module. Code in here: zephyr/scripts/kconfig/kconfiglib.py
During build, e.g. ./zephyrproject/build-icl/Kconfig/Kconfig.modules is created, containing definition for CONFIG_ZEPHYR_SOF_MODULE.

Comment (marc-hb) [reply Kconfig:131]: Thanks, @kv2019i can you please add a reference to `zephyr/modules/Kconfig` in a comment? I think this is not a documented feature so it deserves at least that pointer.

**Comment by marc-hb** [comment]
> yamllint has hung again with no status. I've noticed this on other PRs too.

https://github.com/thesofproject/sof/runs/2602580680 is all green. Please copy the URL when reporting CI issues.

File: Kconfig (hunk 1 above)
Comment (marc-hb) [reply Kconfig:131]: > Thanks, @kv2019i can you please add a reference to zephyr/modules/Kconfig in a comment? 

Submitted in #4327

**Pull request merged**